web/src/components/apps/EmbeddedClusterManagement.tsx

```diff
@@ -457,50 +467,51 @@ const EmbeddedClusterManagement = ({
             />
           </div>
           <p className="tw-text-base tw-text-gray-600">
-            To add a node to this cluster, select the type of node you'd like to
-            add. Once you've selected a node type, we will generate a node join
-            command for you to use in the CLI. When the node successfully joins
-            the cluster, you will see it appear in the list of nodes on this
-            page.
+            {(rolesData?.roles || NODE_TYPES).length > 1 &&
+              "Select one or more roles to assign to the new node. "}
+            Copy the join command and run it on the machine you'd like to join
+            to the cluster.
           </p>
-          <div className="tw-grid tw-gap-2 tw-grid-cols-4 tw-auto-rows-auto">
-            {rolesLoading && (
-              <p className="tw-text-base tw-w-full tw-text-center tw-py-4 tw-text-gray-500 tw-font-semibold">
-                Loading roles...
-              </p>
-            )}
-            {!rolesData && rolesError && (
-              <p className="tw-text-base tw-w-full tw-text-center tw-py-4 tw-text-pink-500 tw-font-semibold">
-                {rolesError?.message}
-              </p>
-            )}
-            {(rolesData?.roles || NODE_TYPES).map((nodeType) => (
-              <div
-                key={nodeType}
-                className={classNames("BoxedCheckbox", {
-                  "is-active": selectedNodeTypes.includes(nodeType),
-                  "is-disabled": determineDisabledState(),
-                })}
-              >
-                <input
-                  id={`${nodeType}NodeType`}
-                  className="u-cursor--pointer hidden-input"
-                  type="checkbox"
-                  name={`${nodeType}NodeType`}
-                  value={nodeType}
-                  disabled={determineDisabledState()}
-                  checked={selectedNodeTypes.includes(nodeType)}
-                  onChange={handleSelectNodeType}
-                />
-                <label
-                  htmlFor={`${nodeType}NodeType`}
-                  className="tw-block u-cursor--pointer u-userSelect--none u-textColor--primary u-fontSize--normal u-fontWeight--medium tw-text-center"
+          {rolesLoading && (
+            <p className="tw-text-base tw-w-full tw-text-center tw-py-4 tw-text-gray-500 tw-font-semibold">
+              Loading roles...
+            </p>
+          )}
+          {!rolesData && rolesError && (
+            <p className="tw-text-base tw-w-full tw-text-center tw-py-4 tw-text-pink-500 tw-font-semibold">
+              {rolesError?.message}
```

Comment:
I think it'd be good to add a fallback error message here if the API doesn't return one for some reason. I'm thinking of something along these lines:
`{rolesError?.message || "Unable to fetch roles"}`